Update `torch.fx.passes.split_module` docstring and class names

patrickspencer opened the pull request:
Summary:
Add docstring for `torch.fx.passes.split_module` that conforms to Google Python Style conventions.

Change `Partition` class to private (by adding underscore) since this was the only file that used that class

Test Plan:
Ran buck build //caffe2:torch to see if the build created any errors caused by the change. No errors detected.

I also ran buck test //caffe2/torch:fx but that didn't have have any test targets.

Differential Revision: D31145694

jamesr66a commented on torch/fx/passes/split_module.py:
```diff
@@ -31,6 +31,67 @@ def split_module(
+        m (GraphModule): Graph module to split
+        root_m (torch.nn.Module): root nn module. Not currently used. Included
+            because the the root nn module is usually transformed via
+            torch.fx._symbolic_trace.symbolic_trace, see example below
```
If this isn't used we should probably just delete it, but we can do it in a follow-up

jamesr66a commented on torch/fx/passes/split_module.py:
```diff
@@ -31,6 +31,67 @@ def split_module(
+            traced = symbolic_trace(mm)
+
+            def split_cb(node: torch.fx.Node) -> int:
+                if node.name == "a" or node.name == "b" or node.name == "add":
```
Node names aren't stable in FX so it's not a good idea to match on them. I suggest changing this to something like:

```
if node.op == 'placeholder' and node.target in {'a', 'b'}:
  return 0
elif node.target is torch.add:
  return 0
else:
  return 1
```

Commit:
Update `torch.fx.passes.split_module` docstring (#65542)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/65542

Add docstring for torch.fx.passes.split_module that conforms to Google Python Style conventions.

Changed original example to the example from this diff:
https://www.internalfb.com/diff/D24925283 (https://github.com/pytorch/pytorch/commit/9734c042b8748c897fa04ded58757c2b66f3b309)

Test Plan:
Ran buck test //caffe2/test:fx. No errors detected
https://pxl.cl/1QCch

Differential Revision: D31145694

fbshipit-source-id: 23ffb0dcb220e053a6621206d94441982b483799

jamesr66a reviewed (approved):
![image](https://user-images.githubusercontent.com/4685384/136134481-1028f084-7aa2-43a5-8715-41a6242964ea.png)

Thanks!
